processor/src/main/resources/org/mapstruct/ap/internal/model/NestedPropertyMappingMethod.ftl

```diff
@@ -12,7 +12,7 @@
     }
 <#list propertyEntries as entry>
     <#if entry.presenceCheckerName?? >
-    if ( !<@localVarName index=entry_index/>.${entry.presenceCheckerName}() ) {
+    if ( <@localVarName index=entry_index/> == null || !<@localVarName index=entry_index/>.${entry.presenceCheckerName}() ) {
```

Comment:
I guess the `||` should be `&&` in order to stay in line with other occurrences.

Reply:
If i change this to &&, it won't fix the null pointer.    This is the code that gets generated for the test case where `hasSourceChild` is `true` and `sourceChild` is `null`

```
 private String sourceParentSourceChildContent(SourceParent sourceParent) {
        if ( sourceParent == null ) {
            return null;
        }
        if ( sourceParent == null || !sourceParent.hasSourceChild() ) {
            return null;
        }
        SourceChild sourceChild = sourceParent.getSourceChild();
        if ( sourceChild == null || !sourceChild.hasContent() ) {
            return null;
        }
        String content = sourceChild.getContent();
        return content;
    }
```

Reply:
Why `hasSourceChild` returns `true` when `sourceChild` is `null`?

Reply:
I missed the discussion in the issue before writing this. @jakraska explained everything in https://github.com/mapstruct/mapstruct/issues/1826#issuecomment-495920222. That is what I suspected as well.

Reply:
> If i change this to &&, it won't fix the null pointer. This is the code that gets generated for the test case where `hasSourceChild` is `true` and `sourceChild` is `null`

Ok. You are right.. You break the flow (return) in stead of "setting some target".. Its de good all " de morgan"  again :smile:.

Reply:
I have only one small remark about how the generated code looks like. The `sourceParent` is checked twice for different to `null`. Maybe we need to skip the second check. I think it can be done with an explicit check for `entry_index == 0`. WDYT @jakraska?

Reply:
@filiphr  - yes i agree with the concern about the duplicate null check.  I left it in there to keep the template logic / change as readable as possible, but i'm definitely happy to add that condition in if you prefer.

Reply:
I updated the template.  the generated code now looks like this:
```
    private String sourceParentSourceChildContent(SourceParent sourceParent) {
        if ( sourceParent == null ) {
            return null;
        }
        if ( !sourceParent.hasSourceChild() ) {
            return null;
        }
        SourceChild sourceChild = sourceParent.getSourceChild();
        if ( sourceChild == null || !sourceChild.hasContent() ) {
            return null;
        }
        String content = sourceChild.getContent();
        return content;
    }
```